src/gtest/test_noteencryption.cpp

```diff
@@ -0,0 +1,170 @@
+#include <gtest/gtest.h>
+#include "sodium.h"
+
+#include <stdexcept>
+
+#include "zcash/NoteEncryption.hpp"
+#include "zcash/prf.h"
+
+class TestNoteDecryption : public ZCNoteDecryption {
+public:
+    TestNoteDecryption(uint256 sk_enc) : ZCNoteDecryption(sk_enc) {}
+
+    void change_pk_enc(uint256 to) {
+        pk_enc = to;
+    }
+};
+
+TEST(noteencryption, api)
+{
+    uint256 sk_enc = ZCNoteEncryption::generate_privkey(uint256S("21035d60bc1983e37950ce4803418a8fb33ea68d5b937ca382ecbae7564d6a77"));
+    uint256 pk_enc = ZCNoteEncryption::generate_pubkey(sk_enc);
+
+    ZCNoteEncryption b = ZCNoteEncryption(uint256());
```

Comment:
Can we add a test that creates two ZCNoteEncryptions and asserts their `.get_epk()` are different values? If I implemented `random_uint256()` to return some constant value like `1` I don't think it would be caught by these tests.